results/cmd/api/main_test.go

```diff
@@ -118,37 +119,287 @@ func TestUpdateTaskRun(t *testing.T) {
+					Name:      "mytaskrun",
+					Namespace: "default",
+				},
+				Spec: &pb.TaskRunSpec{
+					TaskSpec: &pb.TaskSpec{
+						Steps: []*pb.Step{
+							{
+								Name: "new_first_step",
+							},
+							{
+								Name: "new_next_step",
+							},
+						},
+					},
+				},
+			},
+			expect: &pb.TaskRun{
+				ApiVersion: "v1beta1",
+				Metadata: &pb.ObjectMeta{
+					Uid:       "31415926",
+					Name:      "mytaskrun",
+					Namespace: "default",
+				},
+				Spec: &pb.TaskRunSpec{
+					TaskSpec: &pb.TaskSpec{
+						Steps: []*pb.Step{
+							{
+								Name: "new_first_step",
+							},
+							{
+								Name: "new_next_step",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			in: &pb.TaskRun{
+				ApiVersion: "v1alpha1",
+				Metadata: &pb.ObjectMeta{
+					Uid:       "31415926",
+					Name:      "mytaskrun",
+					Namespace: "default",
+				},
+			},
+			name:      "Test Mask with nil Paths field",
+			fieldmask: field_mask.FieldMask{},
+			// do not update
+			update: &pb.TaskRun{
+				ApiVersion: "v1beta1",
+				Metadata: &pb.ObjectMeta{
+					Uid:       "123456",
+					Name:      "newtaskrun",
+					Namespace: "tekton",
+				},
+			},
+			expect: &pb.TaskRun{
+				ApiVersion: "v1alpha1",
+				Metadata: &pb.ObjectMeta{
+					Uid:       "31415926",
+					Name:      "mytaskrun",
+					Namespace: "default",
+				},
+			},
+		},
 	}
-	for rows.Next() {
-		var taskrunblob []byte
-		taskrun := &pb.TaskRun{}
-		rows.Scan(&taskrunblob)
-		if err := proto.Unmarshal(taskrunblob, taskrun); err != nil {
-			t.Fatal("unmarshaling error: ", err)
-		}
-		if diff := cmp.Diff(taskrun.String(), r.GetTaskRun().String()); diff != "" {
-			t.Fatalf("Update Function not properly implemented: %v", diff)
-		}
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			r, err := srv.CreateTaskRunResult(ctx, &pb.CreateTaskRunRequest{TaskRun: tc.in})
+			if err != nil {
+				t.Fatalf("could not create taskrun: %v", err)
+			}
+			// Update the created taskrun
+			fieldMask := &tc.fieldmask
+			if tc.name == "Test no Mask" {
+				fieldMask = nil
+			}
+			r, err = srv.UpdateTaskRunResult(ctx, &pb.UpdateTaskRunRequest{TaskRun: tc.update, ResultsId: r.GetResultsId(), UpdateMask: fieldMask})
+			if err != nil && status.Code(err) != codes.NotFound {
+				t.Fatalf("could not update taskrun: %v, %v", err, status.Code(err))
+			}
+			rows, err := srv.db.Query("SELECT taskrunlog FROM taskrun WHERE results_id = ?", r.GetResultsId())
+			if err != nil {
+				t.Fatalf("failed to query on database: %v", err)
+			}
+			for rows.Next() {
+				var taskrunblob []byte
+				taskrun := &pb.TaskRun{}
+				rows.Scan(&taskrunblob)
+				if err := proto.Unmarshal(taskrunblob, taskrun); err != nil {
+					t.Fatal("unmarshaling error: ", err)
+				}
+				if diff := cmp.Diff(taskrun.String(), tc.expect.String()); diff != "" {
+					t.Fatalf("Update Function not properly implemented: %v, %v, %v", diff, tc.name, tc.fieldmask.String())
+					t.Fatal(tc.name)
```

Comment:
I'm not sure if this will ever print given the fatal right before this. I think this can be removed though since we print the name in the last line.

Reply:
done